nemo/core/optim/adafactor.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Most of the code here has been copied from:
+# https://github.com/pytorch/fairseq/blob/main/fairseq/optim/adafactor.py
+
+import math
+
+import torch
+from torch.optim.optimizer import Optimizer
+
+__all__ = ['Adafactor']
+
+
+class Adafactor(Optimizer):
+    """Implements Adafactor algorithm.
+    This implementation is based on:
+    `Adafactor: Adaptive Learning Rates with Sublinear Memory Cost`
+    (see https://arxiv.org/abs/1804.04235)
+    Note that this optimizer internally adjusts the learning rate
+    depending on the *scale_parameter*, *relative_step* and
+    *warmup_init* options. To use a manual (external) learning rate
+    schedule you should set `scale_parameter=False` and
```

Comment:
A great addition!
One comment: wouldn't the default behaviour here (i.e., using internal scheduler) conflict with the default behaviour of other optimizers (i.e., which rely on external scheduler)? 
Perhaps default should be using external scheduler?